fb_apt/README.md

```diff
@@ -0,0 +1,88 @@
+fb_apt Cookbook
+====================
+The `fb_apt` cookbook installs and configures APT, the Debian package 
+management tool.
+
+Requirements
+------------
+
+Attributes
+----------
+* node['fb_apt']['config']
+* node['fb_apt']['keys']
+* node['fb_apt']['keyring']
+* node['fb_apt']['keyserver']
+* node['fb_apt']['mirror']
+* node['fb_apt']['preserve_sources_list_d']
+* node['fb_apt']['preferences']
+* node['fb_apt']['repos']
+* node['fb_apt']['update_delay']
+* node['fb_apt']['want_backports']
+* node['fb_apt']['want_non_free']
+* node['fb_apt']['want_source']
+
+Usage
+-----
+To install and configure APT include `fb_apt`, which will populate the 
+repository sources in `/etc/apt/sources.list` and update the package cache 
+every `node['fb_apt']['update_delay']` seconds (defaults to 86400) using the 
+`execute[apt-get update]` resource (which other cookbooks can suscribe to or 
+notify against). Note that the `node['fb_apt']['update_delay']` attribute is 
+evaluated at compile time, so you'd need to set it before including `fb_apt` if
+you'd like to customize it.
+
+### Repository sources
+By default the cookbook will setup the base distribution repos based on the
+codename (as defined in `node['lsb']['codename']`). The mirror can be
+customized with `node['fb_apt']['mirror']`; if set to `nil`, base repos will
+not be included at all in `/etc/apt/sources.list`. If base repos are enabled,
+the additional `backports` and `non-free` sources can be enabled with the
+`node['fb_apt']['want_backports']` and `node['fb_apt']['want_non_free']`
+attributes, and source code repos can be enabled with
+`node['fb_apt']['want_source']`; these all default to `false`.
+
+Additiona repository sources can be added with `node['fb_apt']['repos']`. By
+default `fb_apt` will clobber existing contents in `/etc/apt/sources.list.d` to
+ensure it has full control on the repository list; this can be disabled with
+`node['fb_apt']['preserve_sources_list_d']`.
+
+### Keys
+Repository keys can be added to `node['fb_apt']['keys']` which is a hash in the
+`keyid => key` format; if `key` is `nil` the key will be automatically fetched
+from the `node['fb_apt']['keyserver']` keyserver (`keys.gnupg.net` by default).
+Example:
+
+  node.default['fb_apt']['keys'] = {
+    '94558F59' => nil,
+    'F3EFDBD9' => <<-eos
+  -----BEGIN PGP PUBLIC KEY BLOCK-----
+  ...
+  eos
+  }
+
+Automatic key fetching can be disabled by setting the keyserver to `nil`; this 
+will produce an exception for any unspecified key. By default `fb_apt` will 
+manage the keyring at `/etc/apt/trusted.gpg`; this can be customized with
+`node['fb_apt']['keyring']`.
+
+### Configuration
+APT behaviour can be customized using `node['fb_apt']['config']`, which will be
+used to populate `/etc/apt/apt.conf`. Example:
+
+  node.default['fb_apt']['config'] = {
```

Comment:
encouraging people to wholesale overwrite the config is bad. The example should add a config.